modules/core/src/bitgo.ts

```diff
@@ -70,59 +83,54 @@ superagent.Request.prototype.end = function(cb) {
 
 // Handle HTTP errors appropriately, returning the result body, or a named
 // field from the body, if the optionalField parameter is provided.
-superagent.Request.prototype.result = function(optionalField) {
+(superagent as any).Request.prototype.result = function(optionalField?: string) {
   return this.then(handleResponseResult(optionalField), handleResponseError);
 };
 
-const handleResponseResult = function(optionalField: any = false) {
-  return function(res) {
+function handleResponseResult(optionalField?: string): (res: superagent.Response) => any {
+  return function(res: superagent.Response) {
     if (_.isNumber(res.status) && res.status >= 200 && res.status < 300) {
       return optionalField ? res.body[optionalField] : res.body;
     }
     throw errFromResponse(res);
   };
-};
+}
 
-const errFromResponse = function(res) {
+function errFromResponse(res: superagent.Response): Error {
   const errString = createResponseErrorString(res);
   const err: any = new Error(errString);
 
   err.status = res.status;
   if (res.body) {
     err.result = res.body;
   }
-  if (_.has(res.headers, 'x-auth-required') && (res.headers['x-auth-required'] === 'true')) {
+  if (_.has(res.header, 'x-auth-required') && (res.header['x-auth-required'] === 'true')) {
     err.invalidToken = true;
   }
   if (res.body.needsOTP) {
     err.needsOTP = true;
   }
   return err;
-};
+}
 
-const handleResponseError = function(e) {
+function handleResponseError(e): never {
   if (e.response) {
     throw errFromResponse(e.response);
   }
   throw e;
-};
+}
 
 /**
  * There are many ways a request can fail, and may ways information on that failure can be
  * communicated to the client. This function tries to handle those cases and create a sane error string
  * @param res Response from an HTTP request
- * @returns {String}
  */
-const createResponseErrorString = function(res) {
-  let errString = res.statusCode.toString(); // at the very least we'll have the status code
+function createResponseErrorString(res: superagent.Response): string {
+  let errString = res.status.toString(); // at the very least we'll have the status code
   if (res.body.error) {
     // this is the case we hope for, where the server gives us a nice error from the JSON body
     errString = res.body.error;
   } else {
-    // things get messy from here on, we try different parts of the response, salvaging what we can
```

Comment:
why is this deleted?

Reply:
I don't think `res.res` ever exists, so this will never actually run.